<launch>
  <!-- Arguments -->
  <arg name="cmd_vel_topic" default="/robot/cmd_vel" />
  <arg name="odom_topic" default="/odom" />
  <arg name="move_forward_only" default="true"/>
  

  <!-- move_base -->
  <node pkg="move_base" type="move_base" respawn="true" name="move_base" output="screen">
    <param name="base_local_planner" value="dwa_local_planner/DWAPlannerROS" />
    <!-- <param name="base_global_planner"  value="global_planner/GlobalPlanner"/> -->
    <rosparam file="$(find hw_t)/config/costmap_common_param.yaml" command="load" ns="global_costmap" />
    <rosparam file="$(find hw_t)/config/costmap_common_param.yaml" command="load" ns="local_costmap" />
    <rosparam file="$(find hw_t)/config/local_costmap_param.yaml" command="load" />
    <rosparam file="$(find hw_t)/config/global_costmap_param.yaml" command="load" />
    <rosparam file="$(find hw_t)/config/move_base_param.yaml" command="load" />
    <rosparam file="$(find hw_t)/config/dwa_local_planner_param.yaml" command="load" />
    <remap from="cmd_vel" to="$(arg cmd_vel_topic)"/>
    <remap from="/odom/filtered" to="$(arg odom_topic)"/>
    
    <param name="DWAPlannerROS/min_vel_x" value="0.0" if="$(arg move_forward_only)" />
    <param name="TrajectoryPlannerROS/max_vel_theta " value="0.2" />
    <param name="TrajectoryPlannerROS/min_vel_theta " value="0.01" />
    <param name="NavfnROS/use_dijkstra" value="false" />
    <!-- <param name="NavfnROS/use_global_planner/GlobalPlanner" value="true" /> -->
      




  </node>
</launch>
